core/templates/pages/exploration-editor-page/services/exploration-rights.service.ts

```diff
@@ -230,6 +230,14 @@ export class ExplorationRightsService {
   }
 
   removeRoleAsync(memberUsername: string): Promise<void> {
+    const categories = ['ownerNames', 'editorNames', 'viewerNames'] as const;
+    let initialState: {[key: string]: string[]} = {};
```

Comment:
Try picking a different name besides "state", since we already use "state" in the exploration editor to mean something else. Perhaps initialUsernamesByCategory or similar.

That said, I'm a bit confused about why you need to maintain initialState at all. Wouldn't editing this[category] be sufficient?

Reply:
this[category] is supposed to represent this.ownerNames, this.editorNames, etc when doing the lookup in the for loop. The initalUsernamesByCategory is for making a copy of the respective array prior to the deletion, and then using that copy to restore it in the catch block.